include/flatbuffers/flexbuffers.h

```diff
@@ -602,7 +607,7 @@ class Reference {
 
   template<typename T> bool Mutate(const uint8_t *dest, T t, size_t byte_width,
                                    BitWidth value_width) {
-    auto fits = static_cast<size_t>(1U << value_width) <= byte_width;
+    auto fits = static_cast<size_t>(static_cast<size_t>(1U) << value_width) <= byte_width;
```

Comment:
This does not work. VS warns with the following message:
warning C4334: '<<': result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?)

https://msdn.microsoft.com/en-us/library/ke55d167.aspx

We can change 1U to 1u64, but I believe this will cause problems with 32bit builds.